### DISCUSSION TITLE
HEOS Speaker Grouping and Linking Issues

### DISCUSSION THREAD

**Discussion by GizmoGoody** [music-assistant/support#2602] [Q&A] [answered]
I have a HEOS audio system and have pulled 6 of those speakers into MA. Three of them are in the same general area, so I would like to group them as described in Grouping Players method 2 under the [User Interface section of the guide](https://music-assistant.io/ui/).

But, I do not have the ADD GROUP PLAYER option even though all of my Players are HEOS. Why?

![image](https://github.com/music-assistant/hass-music-assistant/assets/46235745/13879781-9505-4991-8ce3-7b7ce9c65f73)

Also, when I attempt to link Speakers (method 1), MA seems to only link 2 of the 3 at a time. Why?

https://github.com/music-assistant/hass-music-assistant/assets/46235745/223887df-97c7-4b98-942d-04f6f93f8ce5

**Comment by GizmoGoody** [comment]
I'll answer my own question. It seems HEOS is not (yet?) natively supported by MA as a player. My HEOS Players are actually just HA entities and as such are not capable of all of the HEOS features, presumably including proper linking and grouping.

More here:
https://github.com/orgs/music-assistant/discussions/1167

**Comment by OzGav** [comment, marked as answer]
You should be able to create a universal group will all of the HEOS players however they won't ncessarily play in sync.

**Reply by GizmoGoody** [reply]
Right, everything else I read indicated those are a last resort. I need the group player to hold a queue, but these speakers are right next to each other, so they must stay in sync.

Is adding HEOS as a native player still in the works?

**Reply by OzGav** [reply]
It is not being worked on at all. It is the most highly voted but a dev with a HEOS amp needs to pick this up

**Reply by GizmoGoody** [reply]
Is there some way I could provide information from my system (two HEOS Drives (Amps) feeding 7 zones and a sound bar) to a dev then work with them to iterate through testing?

With the high polling results, I imagine I wouldn't be the only one interested in helping test even though I lack the coding chops.

**Reply by OzGav** [reply]
There just isn’t anyone with capacity at the moment. Someone new has to come along to work on it.

**Reply by GizmoGoody** [reply]
Ok, well if that changes, let me know and I'm happy to help test.
